lib/member_page.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'scraped'
+require 'pry'
+
+# This class represents the profile page of a given member
+class MemberPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::CleanUrls
+
+  field :id do
+    url.gsub('.htm', '').split('members/').last
```

Comment:
This doesn't really seem like how you want to doing this.

`gsub` implies that you're expecting that there might be more than one instance of the '.htm', and generally isn't really suitable for this sort of thing anyway. Normally you'd just want `File.basename(url, '.htm') here, though this case is a little more awkward because of wanting the 'yrDD' part too.

Building both up here in one place would be a little ugly, but the first part of that is really the `term` field anyway (even if `scraper.rb` is going to want that to be '6' instead, the class should really be exposing what the site claims), so if that already exists, then this becomes something more like:

```ruby
  field :id do
    '%s/%s' % [term, File.basename(url, '.htm')]
  end

  field :term do
    File.dirname(url).split('/').last
  end
```